docs/rfcs/0007-synchronized-releases.md

```diff
@@ -0,0 +1,105 @@
+
+For every GeckoView version we will have a matching Android Components release branch (e.g. `releases/89.0`), that we cut from master the day the GeckoView Nightly version becomes Beta. This branch will continue to track the matching GeckoView version from beta builds to release builds.
+
+### Merge day
+
+On merge day the current Nightly version of GeckoView will become the new Beta version. At this point we will cut a matching Android Components release branch, which will continue to track the GeckoView version, while `master` will move on to track the next Nightly version.
+
+Moving code between components will no longer be needed and the merge day procedure will be reduced to branching and versioning, which will be easier to automate.
+
+Example: On the day GeckoView 89 Nightly becomes GeckoView Beta 89, we will cut an Android Components 89 release branch which will track GeckoView Beta 89 (which will eventually become the release version of GeckoView 89). The `master` branch of Android Components will continue to track GeckoView Nightly 90.
+
+### Automation & Bots
+
+We can continue to bump GeckoView and Android Component versions with our bots. The aligned version numbers simplify the process and even allow us to automate releasing and branching too.
+
+* On `master` a bot will continue to update to the latest GeckoView Nightly version. With this change no other version bumps on `master` are needed.
+* On a release branch (e.g. `releases/89.0`) a bot can update to the latest matching GeckoView version. Over time the branch will be bumped from Beta (e.g. 89.0 Beta) to Release versions of GeckoView (e.g. 89.0 release).
+
+Additionally we could automate:
+
+* Whenever the GeckoView Nightly major version changes on merge day and a new GeckoView Beta is available, a bot could cut a matching Android Components release branch and switch to the GeckoView Beta version (e.g. when GeckoView Nightly 89 switches to Nightly 90 a bot could cut a `releases/89.0` branch, bump it to GeckoView Beta 89 and cut a release).
+
+### Fenix integration
+
+The integration into Fenix would look similar as today:
+* `master` tracks AC Nightly with that GeckoView Nightly
+* A release branch tracks the matching AC version. The aligned versions will avoid confusion (Fenix 89 release branch will use AC 89).
+
+Fenix would still have three different product flavors: Nightly, Beta, Release. Those would continue to exist for supporting different branding and build configurations. The GeckoView release channel and version solely depends and aligns with the Android Components version (Nightly vs. a release version).
+
+In theory release branching (for code freeze and releases) could be automated on the Fenix side too with this change: For every major version bump (AC/GeckoView 89 -> 90) we could automatically cut the release branch and automate updating versions.
+
+### Example
+
+![](/assets/images/rfc/release-trains.png)
```

Comment:
I wonder if integrating GeckoView _Beta_ releases into dot releases `v89.0.1` is confusing. Should we use semantic versioning like Fenix and instead do `v89.0.0-beta.1` until merge day and increment the `beta.X` release for both A-C uplifts and integration of new GV betas?

Then once we hit GV RC we could start using non-Beta releases like `v89.0.1` for A-C uplifts and GV Release Updates.

In the automation around GV integration this is trivial to do. I am not sure about the Task Cluster release building though - we may need to make it aware of `-beta.1` style tags.

Reply:
I agree that this is confusing and mentioned something similar in the `Drawbacks` section. Yes, I think we should consider changing the versioning scheme, but since that requires further release engineering changes (e.g. there are strict assumptions about the versioning scheme right now), I would like to discuss and plan this separately and not make it a requirement for *this* change.

Reply:
That sounds good to me. Let's keep this one simple.